x-pack/plugin/esql/qa/testFixtures/src/main/resources/union_types.csv-spec

```diff
@@ -1351,3 +1351,54 @@ FROM sample_data, sample_data_ts_long
            null | 172.21.0.5   |             1232382 | Disconnected    | Disconnected
            null | 172.21.0.5   |             1232382 | Disconnected    | Disconnected
 ;
+
+shortIntegerWidening
+required_capability: union_types
+required_capability: metadata_fields
+required_capability: casting_operator
+required_capability: union_types_numeric_widening
+
+FROM apps, apps_short METADATA _index
+| EVAL id = id::integer
```

Comment:
This is cool!

However, to reduce risk some more (some data types can be quirky and stuff could happen at block loading time, I guess, to types like `float` and `byte` ...); how about we add to this/replace this by a parameterized integration test that exercises all widened data types?

RestEsqlTestCase is quite nice and, like csv tests, is executed in multiple environments (single/multi node etc.).

Reply:
Tests in `RestEsqlTestCase` also have the benefit that we can more easily test cases with more than 2 different types becoming a union type. E.g. `float`, `byte` and, I dunno, `keyword` being stuffed into `to_string`.